### PR TITLE
[IOS-57]: Algorithm: arithmetic progression, [IOS-59]: Algorithm: geometric progression

### PR DESCRIPTION
**Pull request by zhenyapavlovsky**
Algorithm: arithmetic progression

- Task 1 complete
- Task 2 complete
- Task 3 complete
- Task 4 complete
- Task 5 complete

Algorithm: geometric progression

- Task 1 complete
- Task 2 complete
- Task 3 complete
- Task 4 complete
- Task 5 complete

### CODE REVIEW

File: AlgoritmHomeWork.playground/Contents.swift [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    var i = 0
+    var sum = 0
+    var number = 1
+
+    while i < n {
+        sum = number  + sum
+        number = number * 2
+        i = i + 1
+    }
+
```
Comment (romansorochak) [inline AlgoritmHomeWork.playground/Contents.swift:17]: remove empty line

File: AlgoritmHomeWork.playground/Contents.swift [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+ return sum
```
Comment (romansorochak) [inline AlgoritmHomeWork.playground/Contents.swift:18]: bad alignment - missing tab (some spaces)  from left

File: AlgoritmHomeWork.playground/Contents.swift [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+
+func function2(n: Int) -> Int {
+  guard n > 0 else { return 0 }
+    var i = 0
+    var sum = 0
+    var number = 1
+
+    while i < n {
+
```
Comment (romansorochak) [inline AlgoritmHomeWork.playground/Contents.swift:34]: remove empty line

File: AlgoritmHomeWork.playground/Contents.swift [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        number = number * 3
+        sum = number + sum
+        i = i + 1
+    }
+
```
Comment (romansorochak) [inline AlgoritmHomeWork.playground/Contents.swift:39]: remove empty line

File: AlgoritmHomeWork.playground/Contents.swift [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+ return sum
```
Comment (romansorochak) [inline AlgoritmHomeWork.playground/Contents.swift:40]: bad alignment

File: AlgoritmHomeWork.playground/Contents.swift [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+let kk = function2(n: 2)
+
+
+// Task: 3 Calculate sum: 200 + 100 + 50 + 25 + 12,5 + 6,25 + ... , where n is count of numbers
+
+func function3(n: Int) -> Double {
+  guard n > 0 else { return 0 }
```
Comment (romansorochak) [inline AlgoritmHomeWork.playground/Contents.swift:49]: bad alignment

File: AlgoritmHomeWork.playground/Contents.swift [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+    var i = 0
+    var sum = 0
+    var number = Double(200)
+
+    while i < n {
+        sum = Int(number) + sum
+        number = number / 2
+        i = i + 1
+    }
+
```
Comment (romansorochak) [inline AlgoritmHomeWork.playground/Contents.swift:59]: remove empty line

File: AlgoritmHomeWork.playground/Contents.swift [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+    var i = 0
+    var sum = 0
+    var number = 5
+
+    while i < n {
+        sum = number + sum
+        number = number * (-2)
+        i = i + 1
+    }
+
```
Comment (romansorochak) [inline AlgoritmHomeWork.playground/Contents.swift:79]: remove empty line

File: AlgoritmHomeWork.playground/Contents.swift [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+    return Double(sum)
+}
+
+let kkk = function3(n: 4)
+
+
+// Task: 4 Calculate sum: 5 - 10 + 20 - 40 + 80 - 160 + ... , where n is count of numbers
+
+func function4(n: Int) -> Int {
+  guard n > 0 else { return 0 }
```
Comment (romansorochak) [inline AlgoritmHomeWork.playground/Contents.swift:69]: bad alignment

File: arithmeticAlgoritm.playground/Contents.swift [hunk 10]
```diff
@@ -0,0 +1,104 @@
+import UIKit
+
+// Task:1    Calculate sum: 1 + 2 + 3 + 4 + 5 + 6 + 7 + ..., where n is count of numbers
+
+
+func function1(n: Int) -> Int {
+  guard n > 0 else { return 0 }
```
Comment (romansorochak) [inline arithmeticAlgoritm.playground/Contents.swift:7]: bad alignment

File: arithmeticAlgoritm.playground/Contents.swift [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+    var i = 0
+    var sum = 0
+    var number = 0
+
+    while i < n {
+        number = number + 1
+        sum = number  + sum
+        i = i + 1
+    }
+
```
Comment (romansorochak) [inline arithmeticAlgoritm.playground/Contents.swift:17]: remove empty line

File: arithmeticAlgoritm.playground/Contents.swift [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+ return sum
```
Comment (romansorochak) [inline arithmeticAlgoritm.playground/Contents.swift:18]: bad alignment

File: arithmeticAlgoritm.playground/Contents.swift [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+    var sum = 0
+    var number = 1
+
+    while i < n {
+        sum = number  + sum
+        number = number + 2
+        i = i + 1
+    }
+
+ return sum
```
Comment (romansorochak) [inline arithmeticAlgoritm.playground/Contents.swift:38]: bad alignment

File: arithmeticAlgoritm.playground/Contents.swift [hunk 14]
```diff
@@ -0,0 +1,104 @@
...
+    var i = 0
+    var sum = 0
+    var number = 1
+
+    while i < n {
+        sum = number  + sum
+        number = number + 2
+        i = i + 1
+    }
+
```
Comment (romansorochak) [inline arithmeticAlgoritm.playground/Contents.swift:37]: bad alignment
remove empty line

File: arithmeticAlgoritm.playground/Contents.swift [hunk 15]
```diff
@@ -0,0 +1,104 @@
...
+}
+
+let kk = function2(n: 6)
+
+
+//Task:2    Calculate sum: 5 - 4 - 3 - 2 - 1 - 0 - (-1) - (-2) - ... , where n is count of numbers
+
+
+func function3(n: Int) -> Int {
+
```
Comment (romansorochak) [inline arithmeticAlgoritm.playground/Contents.swift:48]: remove empty line

File: arithmeticAlgoritm.playground/Contents.swift [hunk 16]
```diff
@@ -0,0 +1,104 @@
...
+    var i = 0
+    var sum = 10
+    var number = 5
+
+    while i < n {
+        sum = sum  - number
+        number = number - 1
+        i = i + 1
+    }
+
```
Comment (romansorochak) [inline arithmeticAlgoritm.playground/Contents.swift:58]: remove empty line

File: arithmeticAlgoritm.playground/Contents.swift [hunk 17]
```diff
@@ -0,0 +1,104 @@
...
+ return sum
```
Comment (romansorochak) [inline arithmeticAlgoritm.playground/Contents.swift:59]: bad alignment

File: arithmeticAlgoritm.playground/Contents.swift [hunk 18]
```diff
@@ -0,0 +1,104 @@
...
+}
+
+let kkk = function3(n: 1)
+
+
+
+// Task: 4 Calculate sum: 100 - 95 - 90 - 85 - ... , where n is count of numbers
+
+func function4(n: Int) -> Int {
+  guard n > 0 else { return 0 }
```
Comment (romansorochak) [inline arithmeticAlgoritm.playground/Contents.swift:69]: bad alignment

File: arithmeticAlgoritm.playground/Contents.swift [hunk 19]
```diff
@@ -0,0 +1,104 @@
...
+    var i = 0
+    var sum = 200
+    var number = 100
+
+    while i < n {
+        sum = sum - number
+        number = number - 5
+        i = i + 1
+    }
+
```
Comment (romansorochak) [inline arithmeticAlgoritm.playground/Contents.swift:79]: remove empty line

**Review by romansorochak — CHANGES_REQUESTED** [review]

File: AlgoritmHomeWork.playground/Contents.swift (hunk 1 above)
Comment (zhenyapavlovsky) [reply AlgoritmHomeWork.playground/Contents.swift:17]: Done

File: AlgoritmHomeWork.playground/Contents.swift (hunk 2 above)
Comment (zhenyapavlovsky) [reply AlgoritmHomeWork.playground/Contents.swift]: done

File: AlgoritmHomeWork.playground/Contents.swift (hunk 6 above)
Comment (zhenyapavlovsky) [reply AlgoritmHomeWork.playground/Contents.swift]: done

File: AlgoritmHomeWork.playground/Contents.swift (hunk 7 above)
Comment (zhenyapavlovsky) [reply AlgoritmHomeWork.playground/Contents.swift]: done

File: AlgoritmHomeWork.playground/Contents.swift (hunk 9 above)
Comment (zhenyapavlovsky) [reply AlgoritmHomeWork.playground/Contents.swift]: done

File: arithmeticAlgoritm.playground/Contents.swift (hunk 10 above)
Comment (zhenyapavlovsky) [reply arithmeticAlgoritm.playground/Contents.swift]: done

File: arithmeticAlgoritm.playground/Contents.swift (hunk 11 above)
Comment (zhenyapavlovsky) [reply arithmeticAlgoritm.playground/Contents.swift]: done

File: arithmeticAlgoritm.playground/Contents.swift (hunk 12 above)
Comment (zhenyapavlovsky) [reply arithmeticAlgoritm.playground/Contents.swift]: done

File: arithmeticAlgoritm.playground/Contents.swift [hunk 20]
```diff
@@ -0,0 +1,104 @@
...
+ return sum
+}
+
+let k = function1(n: 5)
+
+// Task:2    Calculate sum: 1 + 3 + 5 + 7 + ..., where n is count of numbers
+
+
+func function2(n: Int) -> Int {
+  guard n > 0 else { return 0 }
```
[the remark this answers is not in the document]
Comment (zhenyapavlovsky) [reply arithmeticAlgoritm.playground/Contents.swift]: done

File: arithmeticAlgoritm.playground/Contents.swift (hunk 13 above)
Comment (zhenyapavlovsky) [reply arithmeticAlgoritm.playground/Contents.swift]: done

File: arithmeticAlgoritm.playground/Contents.swift (hunk 16 above)
Comment (zhenyapavlovsky) [reply arithmeticAlgoritm.playground/Contents.swift]: done

File: arithmeticAlgoritm.playground/Contents.swift (hunk 17 above)
Comment (zhenyapavlovsky) [reply arithmeticAlgoritm.playground/Contents.swift]: done

File: arithmeticAlgoritm.playground/Contents.swift (hunk 18 above)
Comment (zhenyapavlovsky) [reply arithmeticAlgoritm.playground/Contents.swift]: done

**Review by romansorochak — APPROVED** [review]

**Pull request merged**